apps/web/src/server/applications/lib/services/case.service.js

```diff
@@ -93,11 +94,59 @@ export const updateCase = async (caseId, payload) => {
 };
 
 /**
- * Get documents categories for the case
+ * Get all the subfolders in a folder, or the top level folders for the case
  *
  * @param {number} id
+ * @param {number | null} folderId
  * @returns {Promise<DocumentationCategory[]>}
  */
-export const getCaseDocumentationCategories = (id) => {
-	return get(`applications/${id}/documents`);
+export const getCaseFolders = (id, folderId = null) => {
+	return folderId
+		? get(`applications/${id}/folders/${folderId}/sub-folders`)
+		: get(`applications/${id}/folders`);
+};
+
+/**
+ * Get a single folder on a case
+ *
+ * @param {number} caseId
+ * @param {number} id
+ * @returns {Promise<DocumentationCategory>}
+ */
+export const getCaseFolder = (caseId, id) => {
+	const folder = get(`applications/${caseId}/folders/${id}`);
+
+	return folder;
+};
+
+/**
+ * Get the path of all parent folder(s) for the current folder - used for breadcrumbs etc.
+ * returned as an ordered list
+ *
+ * @param {number} caseId
+ * @param {number} folderId
+ * @returns {Promise<DocumentationCategory[]>}
+ */
+export const getCaseDocumentationFolderPath = (caseId, folderId) => {
+	const folderPath = get(`applications/${caseId}/folders/${folderId}/parent-folders`);
+
+	return folderPath;
+};
+
+/**
+ * Get the documents for the current folder
+ *
+ * @param {number} caseId
+ * @param {DocumentationCategory |undefined} documentationCategory
+ * @returns {DocumentationFile[]}
+ */
+export const getCaseDocumentationFilesInFolder = (caseId, documentationCategory) => {
+	// TODO: Mock Version - to be replaced when API to get documents exists:
+	/** @type {DocumentationFile[] } */
+	let documentationFiles = [];
+
+	if (caseId && documentationCategory) {
+		documentationFiles = [{ fileName: 'sitting-1.png', url: '#' }];
+	}
+	return documentationFiles;
```

Comment:
also, turns out there's a lot of services specific of the documentation section.
Instead of putting them in the general case.service file they might sit on their own pages/case/documentation/applications-documentation.service.js, next to the controller and the router.

what do you think?

Reply:
I agree